update_data.bash

```diff
@@ -16,29 +17,35 @@ fi
 function metadata_url {
 	local os=$1
 	local arch=$2
+	local release=$3
 
-	echo "https://joschi.github.io/java-metadata/metadata/ga/${os}/${arch}.json"
+	echo "https://joschi.github.io/java-metadata/metadata/${release}/${os}/${arch}.json"
 }
 
 function fetch_metadata {
 	local os=$1
 	local arch=$2
-	local url
-	url=$(metadata_url "$os" "$arch")
+	local release=$3
 
 	local args=('-s' '-f' '--compressed' '-H' "Accept: application/json")
 	if [[ -n "${GITHUB_API_TOKEN:-}" ]]; then
 		args+=('-H' "Authorization: token $GITHUB_API_TOKEN")
 	fi
 
-	curl "${args[@]}" -o "${DATA_DIR}/jdk-${os}-${arch}.json" "${url}"
+	local url
+	url=$(metadata_url "$os" "$arch" "$release")
+	curl "${args[@]}" -o "${DATA_DIR}/jdk-${os}-${arch}-${release}.json" "${url}"
 }
 
 for OS in $LIST_OS
 do
 	for ARCH in $LIST_ARCH
 	do
-		fetch_metadata "$OS" "$ARCH"
+		for RELEASE_TYPE in $LIST_RELEASE_TYPE
+		do
+			fetch_metadata "$OS" "$ARCH" "$RELEASE_TYPE"
+		done
+		cat "${DATA_DIR}/jdk-${OS}-${ARCH}"-*.json | jq -s 'add' > "${DATA_DIR}/jdk-${OS}-${ARCH}.json"
```

Comment:
This means that old clients (i. e. old versions of the `asdf-java` plugin) would receive the "all" version which would suddenly lead to an "explosion" of installation candidates.

Maybe the merged data file could have an "-all" prefix?

Reply:
👍🏼 Fixed, we have four files: the one without the release type for backward compatibility (same as ga), ea, ga, all.